cores/arduino/stm32/rtc.h

```diff
@@ -0,0 +1,181 @@
+  *      may be used to endorse or promote products derived from this software
+  *      without specific prior written permission.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  *
+  ******************************************************************************
+  */
+
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef __RTC_H
+#define __RTC_H
+
+/* Includes ------------------------------------------------------------------*/
+#include "stm32_def.h"
+
+#ifdef HAL_RTC_MODULE_ENABLED
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+/* Exported types ------------------------------------------------------------*/
+typedef enum {
+  HOUR_FORMAT_12,
+  HOUR_FORMAT_24
+} hourFormat_t;
+
+typedef enum {
+  AM,
+  PM
+} hourAM_PM_t;
+
+// Clock source selection
+typedef enum {
+  LSI_CLOCK,
+  LSE_CLOCK,
+  HSE_CLOCK
+} sourceClock_t;
+
+typedef void(*voidCallbackPtr)(void *);
+
+/* Exported constants --------------------------------------------------------*/
+#if defined(STM32F0xx) || defined(STM32L0xx)
+#define RTC_Alarm_IRQn RTC_IRQn
+#define RTC_Alarm_IRQHandler  RTC_IRQHandler
+#endif
+
+#if defined(STM32F1xx) && !defined(IS_RTC_WEEKDAY)
+// Compensate missing HAL definition
+#define IS_RTC_WEEKDAY(WEEKDAY) (((WEEKDAY) == RTC_WEEKDAY_MONDAY)    || \
+                                 ((WEEKDAY) == RTC_WEEKDAY_TUESDAY)   || \
+                                 ((WEEKDAY) == RTC_WEEKDAY_WEDNESDAY) || \
+                                 ((WEEKDAY) == RTC_WEEKDAY_THURSDAY)  || \
+                                 ((WEEKDAY) == RTC_WEEKDAY_FRIDAY)    || \
+                                 ((WEEKDAY) == RTC_WEEKDAY_SATURDAY)  || \
+                                 ((WEEKDAY) == RTC_WEEKDAY_SUNDAY))
+
+#define IS_RTC_HOUR12(HOUR)      IS_RTC_HOUR24(HOUR)
+#endif // defined(STM32F1xx) && !defined(IS_RTC_WEEKDAY)
+
+/* RTC prescalers are set to obtain the RTC clock to 1Hz. See AN4759.
+  Default prescalers are calculated for the following clock speed:
+  - LSI = 32, 37 or 40 kHz
+  - LSE = 32.768 kHz
+  - HSE = 8 MHz
+
+ Prescalers should be set by user in case the clock configuration is different.
```

Comment:
It should be possible to define [HSE|LSE|LSI]_A?SYNC_PREDIV as for each variant clock values are defined.
Ex: https://github.com/stm32duino/Arduino_Core_STM32/blob/2415879ab2472618918a8f2ca054996d54cb21d4/variants/NUCLEO_L476RG/stm32l4xx_hal_conf.h#L159